rfcs/000-resource-notices.md

```diff
@@ -0,0 +1,88 @@
+- Feature Name: Resource Notives
+- Start Date: 2021-03-22
+- RFC PR: [amundsen-io/rfcs#0000](https://github.com/amundsen-io/rfcs/pull/0000) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Resource Notices
+
+## Summary
+
+Resource Notices allows Amundsen administrators to show informational, warning and alert messages related to the different resources (tables, dashboards, people) we expose in Amundsen
+
+## Motivation
+
+This feature will help administrators to show messages related to deprecation, updates (or lack of), and informational messages that are related to specific resources. In Lyft's case, we are using them to point final users to a improved versions of current surfaced tables, so final users employ more trusted data.
+
+## Guide-level Explanation (aka Product Details)
+
+An Amundser final user can see a notice in the table/dashboard/user detail page whenever his system admin adds one related to it.
+
+A notice is a small box with an icon and a message that can contain HTML markup (like links and bolded text).
+
+These will come in three flavors:
+
+- Informational: Marked with a blue "i" icon on the right side
+- Warning: Marked with an orange exclamation mark icon on the right side
+- Alert: Marked with a red exclamation mark icon on the right side
+
+For example, if company X wants to deprecate the use of one table, they can opt to add a new notice in their configuration file:
+
+```
+    ... Table Configuration
+    notices: {
+        "<SCHEMANAME>.<TABLENAME>": {
+          severity: NoticeSeverity.ALERT,
+          message: `<span>This table is deprecated, please use <a href="<LINKTONEWTABLEDETAILPAGE>">this new table</a> instead.</span>`,
+        },
+    },
+```
+
+This will show a notice with a red exclamation icon whenever a final user visits the Table Detail page of this table.
+
+The ultimate goal of this feature is to allow Amundsen administrators to point their users to more trusted/higher quality data without removing the old references.
+
+## Reference-level Explanation (aka Technical Details)
+
+The design we propose is based in the following:
+
+- Using our current frontend configuration files (config-utils.ts, config-default.ts, config-custom.ts and config-types.ts)
+- Allowing a three level "severity" schema for the alerts (info, warning, and alert)
+- Allowing its use in "tables" and "dashboards" (we could add it to people as well, but we'll need designs for it)
+
+What we are not supporting for now:
+
+- Multiple notifications per resource
```

Comment:
I think we should support this from the get-go: it will make it far easier to have multiple sources of notices in the future if everything is array-aware in the UI. Thoughts?

Reply:
The main problem we see is how do we limit the notifications we show.

Do we set an arbitrary limit like three? Or should we add a "see more" link that expands? Or a modal?

I would be OK to add it if we limit them with an arbitrary threshold, but the rest seem too much thinking in advance for us right now!

Reply:
I sorta think it's the responsibility of the config writers to restrain themselves in how many they include. that said, hard limiting to 2 or 3 is eminently reasonable.

Reply:
I suggest using recency, i.e. we show up to the 3 latest notices

Reply:
the notices config doesn't have dates associated with them... so I think priority sort is only thing possible without changing that spec?

Reply:
True, there are no dates and we cannot ensure the position of elements on the object as we render them.

Reply:
I am also in favor of `no-limits`. True, users might create a lot of them in graph, but it is the same case with Programmatic Descriptions. It's up to end users to say how many of them are required. We could enforce sorting, so the errors are always on top, later there are warnings and info are on the bottom.

Reply:
That's a good point @mgorsk1. I also think sorting by severity makes sense.